odl/space/base_tensors.py

```diff
@@ -866,27 +759,37 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
 
         # --- Process `inputs` --- #
 
-        # Convert inputs that are ODL tensors to Numpy arrays so that the
-        # native Numpy ufunc is called later
+        # Convert inputs that are ODL tensors or their data containers to
+        # Numpy arrays so that the native Numpy ufunc is called later
         inputs = tuple(
-            inp.asarray() if isinstance(inp, type(self)) else inp
+            np.asarray(inp) if isinstance(inp, (type(self), type(self.data)))
+            else inp
             for inp in inputs)
 
         # --- Get some parameters for later --- #
 
         # Arguments for `writable_array` and/or space constructors
-        order = str(kwargs.get('order', self.order)).upper()
-        array_kwargs = {'order': order}
-        order = 'A' if order == 'K' else order  # We don't use 'K'
-
         out_dtype = kwargs.get('dtype', None)
-        if out_dtype is not None:
-            array_kwargs['dtype'] = out_dtype
+        if out_dtype is None:
+            array_kwargs = {}
+        else:
+            array_kwargs = {'dtype': out_dtype}
 
         # --- Evaluate ufunc --- #
 
-        # Trivial context to be used when an output is `None`
+        # Trivial context used to create a single code path for the ufunc
```

Comment:
This should be moved to `odl.util`

Reply:
I have my doubts. Where would this ever be useful?